### PR TITLE
Fix release build errors and copy files from wrapper package

### PR DESCRIPTION
**Pull request by jryckman**
# Justification
The first submission got the debug builds to run and unit tests to pass, however release builds were failing with roslyn errors. This was because of the improper configuration/existence of ruleset files. By default, pulling in a package reference, which was done in Directory.build.props, will automatically make those analyzers run. Those analyzers are not in sync with our coding conventions and NI.rulset. I needed to make everything work to get release builds working and at least fix any existing code analysis warnings to enable a release build.

# Implementation
I deleted the old ruleset files that didn't exist and imported the NI.ruleset (and other rulesets that we use in our analyzer wrapper package). I fixed any existing release failures to get a release build working with tests passing. I currently do have not have a package configured, but that would be the next logical step. The analyzers we're building are also not being consumed in the ruleset, but we have to wait until we can first build a package that we can consume internally in this project.

# Testing
Ran all unit tests and built debug and release.

### CODE REVIEW

**Review by Everest2020 — APPROVED** [review]

File: src/NationalInstruments.Analyzers/Style/SpellingAnalyzer.cs [hunk 1]
```diff
@@ -276,50 +276,6 @@ private static CodeAnalysisDictionary ParseDicDictionary(SourceText text)
         private static string RemovePrefixIfPresent(string prefix, string name)
             => name.StartsWith(prefix, StringComparison.Ordinal) ? name.Substring(1) : name;
 
-        private static IEnumerable<Diagnostic> GetUnmeaningfulIdentifierDiagnostics(ISymbol symbol, string symbolName)
```
Comment (pakdev) [inline src/NationalInstruments.Analyzers/Style/SpellingAnalyzer.cs:279]: Unused, I assume?

File: rulesets/NI.TestUtilities.ruleset [hunk 2]
```diff
@@ -0,0 +1,156 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="NI.Tests.ruleset" Description="Code analysis rule overrides for NI Test projects." ToolsVersion="14.0">
```
Comment (pakdev) [inline rulesets/NI.TestUtilities.ruleset]: Update Name

File: rulesets/NI.MinimumRecommendedRules.ruleset [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="NI.ruleset" Description="Code analysis rules for NI projects." ToolsVersion="14.0">
```
Comment (pakdev) [inline rulesets/NI.MinimumRecommendedRules.ruleset]: Update Name

File: rulesets/NI.Illegal.ruleset [hunk 4]
```diff
@@ -0,0 +1,156 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="NI.Tests.ruleset" Description="Code analysis rule overrides for NI Test projects." ToolsVersion="14.0">
```
Comment (pakdev) [inline rulesets/NI.Illegal.ruleset]: Update Name

File: configuration/NI.Analysis.targets [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  <ItemGroup Condition="'$(IncludeCodeAnalyzers)' == 'True'">
+    <RoslynAnalyzer Include="Microsoft.CodeQuality" />
+    <RoslynAnalyzer Include="Microsoft.NetCore" />
+    <RoslynAnalyzer Include="Microsoft.NetFramework" />
+    <RoslynAnalyzer Include="Text" />
+
+    <!-- Microsoft's CA analyzers -->
+    <Analyzer Include="@(RoslynAnalyzer->'$(NuGetPackagesDirectory)\%(Identity).analyzers\2.6.3\analyzers\dotnet\cs\%(Identity).Analyzers.dll')"/>
+    <Analyzer Include="@(RoslynAnalyzer->'$(NuGetPackagesDirectory)\%(Identity).analyzers\2.6.3\analyzers\dotnet\cs\%(Identity).CSharp.Analyzers.dll')"/>
+    <Analyzer Include="$(PkgMicrosoft_VisualStudio_Threading_Analyzers)\analyzers\cs\Microsoft.VisualStudio.Threading.Analyzers.dll"/>
```
Comment (pakdev) [inline configuration/NI.Analysis.targets]: Where is `PkgMicrosoft_VisualStudio_Threading_Analyzers`/`PkgMicrosoft_CodeQuality_Analyzers` defined?

File: configuration/NI.Analysis.targets [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      <Link>$(CodeAnalysisRuleSet)</Link>
+    </AdditionalFiles>
+
+    <!-- Microsoft's SA analyzers -->
+    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.dll"/>
+    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll"/>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(IncludeCodeAnalyzers)' == 'True'">
+    <RoslynAnalyzer Include="Microsoft.CodeQuality" />
```
Comment (pakdev) [inline configuration/NI.Analysis.targets]: Explicitly adding Microsoft's analyzers is probably unnecessary (check Install.ps1)

File: configuration/NI.Analysis.targets [hunk 7]
```diff
@@ -0,0 +1,95 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <CodeAnalysisRuleSetDirectory>$(PkgNI_DotNet_Analyzers)\content</CodeAnalysisRuleSetDirectory>
```
Comment (pakdev) [inline configuration/NI.Analysis.targets]: Are we assuming clients define PkgNI_DotNet_Analyzers?

File: src/NationalInstruments.Analyzers/Style/SpellingAnalyzer.cs (hunk 1 above)
Comment (jryckman) [reply src/NationalInstruments.Analyzers/Style/SpellingAnalyzer.cs:279]: Yes, sorry I didn't comment on that. It was an unused method.

File: rulesets/NI.MinimumRecommendedRules.ruleset (hunk 3 above)
Comment (jryckman) [reply rulesets/NI.MinimumRecommendedRules.ruleset]: I'm actually going to remove this file until a subsequent submission

File: rulesets/NI.Illegal.ruleset (hunk 4 above)
Comment (jryckman) [reply rulesets/NI.Illegal.ruleset]: Removing this file

File: configuration/NI.Analysis.targets (hunk 5 above)
Comment (jryckman) [reply configuration/NI.Analysis.targets]: I'm removing this file for now, but those variables are generated with <GeneratePathProperty> in the package references that include those packages. We'll revisit this one when we attempt to add it.

File: configuration/NI.Analysis.targets (hunk 6 above)
Comment (jryckman) [reply configuration/NI.Analysis.targets]: Removing this file for now, but we should revisit this for sure

File: configuration/NI.Analysis.targets (hunk 7 above)
Comment (jryckman) [reply configuration/NI.Analysis.targets]: This variable is defined through a <GeneratePathProperty> but I am removing this file for now

File: rulesets/NI.TestUtilities.ruleset (hunk 2 above)
Comment (jryckman) [reply rulesets/NI.TestUtilities.ruleset]: Removed this file

**Pull request merged**